ipn/ipnlocal/cert.go

```diff
@@ -458,14 +468,26 @@ func acmeKey(dir string) (crypto.Signer, error) {
 	return privKey, nil
 }
 
-func validCertPEM(domain string, keyPEM, certPEM []byte, now time.Time) bool {
+func validCertPEM(domain string, keyPEM, certPEM, extraRoots []byte, now time.Time) bool {
 	if len(keyPEM) == 0 || len(certPEM) == 0 {
 		return false
 	}
 	tlsCert, err := tls.X509KeyPair(certPEM, keyPEM)
 	if err != nil {
 		return false
 	}
+	roots, err := x509.SystemCertPool()
+	if err != nil {
+		log.Printf("cert check: failed to get system roots: %v", err)
+		return false
+	}
+	if len(extraRoots) != 0 {
+		if !roots.AppendCertsFromPEM(extraRoots) {
+			log.Print("cert check: failed append certificates")
+			return false
+		}
+	}
+
```

Comment:
instead of appending to the system `*x509.CertPool` here, create an empty root for in the test code and pass that directly

Reply:
Oh, I see what you are saying now. Will update again.

Reply:
Done, you're right this is better.